src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -423,12 +411,14 @@ void primitive_inst::update_shape() {
             auto data_padding = padding::max(_impl_params->get_output_layout(idx).data_padding, layout.data_padding);
             layout.data_padding = padding::max(_node->get_primitive()->get_output_padding(idx), data_padding);
         }
-        if (_impl_params->get_output_layout(idx) != layout) {
+        auto& old_layout = _impl_params->output_layouts[idx];
+        // check only paddings since only they are changed
```

Comment:
This function is used to assign newly recalculated layout to _impl_params; it's not about paddings only. Therefore `_impl_params->output_layouts[idx] = layout;` should be called not only if padding was changed